### PR TITLE
Raw initial

### PR DESCRIPTION
**Pull request by LoganWright**

### CODE REVIEW

File: Sources/Fluent/Memory/Memory+Filters.swift [hunk 1]
```diff
@@ -148,6 +148,10 @@ extension Node {
             case .or:
                 return passesOne(filters)
             }
+        case .raw(command: let command, values: let values):
+            // TODO: Throw error, memory DB doesn't support raw?
```
Comment (tannernelson) [inline Sources/Fluent/Memory/Memory+Filters.swift]: Yeah we should definitely throw an error. Also mongodb isn't going to support raw.

File: Sources/Fluent/Query/Filter.swift [hunk 2]
```diff
@@ -129,3 +132,15 @@ extension QueryRepresentable {
         return try filter(T.self, field, .contains, value)
     }
 }
+
+extension QueryRepresentable {
+    @discardableResult
+    public func raw(command: String, values: [NodeRepresentable] = []) throws -> Query<Self.T> {
```
Comment (tannernelson) [inline Sources/Fluent/Query/Filter.swift]: We'll need to check with @Casperhr to make sure this completely solves what he's wanting.

I know in Laravel there was strange things like:

```php
User::where('timestamp', '>', DB::raw("TIMESTAMP()"))->all()
```

This basically allows you to bypass the ` escape that happens to strings that are passed.

I don't know if that's some we want or need or if this level of granularity is fine.

Comment (LoganWright) [inline Sources/Fluent/Query/Filter.swift]: I don't fully understand this comment, so @Casperhr please weigh in if this is the type of level you're looking for.

File: Sources/Fluent/Memory/Memory+Filters.swift (hunk 1 above)
Comment (LoganWright) [inline Sources/Fluent/Memory/Memory+Filters.swift]: The reason I thought maybe it shouldn't throw an error is if users want to use raw in some way w/ memory db just for testing, but I guess this wouldn't work either way, unless we want to build a SQL interpreter ... which I do not, so throwing is probably the expectation

File: Sources/Fluent/Database/ThreadConnectionPool.swift [hunk 3]
```diff
@@ -2,87 +2,125 @@ import Foundation
...
+            is no longer capable of creating new ones.
+        */
+        case maxConnectionsReached(max: Int)
+
+        /**
+            This is here to allow extensibility w/o breaking apis, it is not currently
+            used, but should be accounted for by end user if they are handling the 
+            error
+        */
+        case open(Swift.Error)
```
Comment (tannernelson) [inline Sources/Fluent/Database/ThreadConnectionPool.swift]: I'm on the fence about the name `.open`. If I were using this framework and I saw that in a switch statement I would think it has something to do with opening connections. `.unknown` is unambiguous.

File: Package.swift [hunk 4]
```diff
@@ -8,9 +8,9 @@ let package = Package(
     ],
     dependencies: [
         // Data structure for converting between multiple representations
-        .Package(url: "https://github.com/vapor/node.git", Version(0,0,0)),
+        .Package(url: "https://github.com/vapor/node.git", Version(2,0,0, prereleaseIdentifiers: ["alpha.1"])),
```
Comment (tannernelson) [inline Package.swift]: Should we rely on just `"alpha"` here?

**Pull request merged**